### PR TITLE
fix(client): add tests for layoutSelector

### PR DESCRIPTION
**Pull request by ahmaxed**

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
`react-i18next` does not include  `initReactI18next` when running tests; therefore, I get the following error when using the `configForTest.js`: 

 `You are passing an undefined module! Please check the object you are passing to i18next.use()
However it does when developing.`

These are the only keys that `react-i18next` include: 
```
    {
      withTranslation: [Function: withTranslation],
      useTranslation: [Function: useTranslation],
      Trans: [Function: Trans],
      default: {
        withTranslation: [Function: withTranslation],
        useTranslation: [Function: useTranslation],
        Trans: [Function: Trans]
      }
    }
```

However, when developing and logging `react-i18next` includes `initReactI18next` among other methods as seen below:
```
{
  Trans: [Getter],
  useTranslation: [Getter],
  withTranslation: [Getter],
  Translation: [Getter],
  I18nextProvider: [Getter],
  withSSR: [Getter],
  useSSR: [Getter],
  I18nContext: [Getter],
  initReactI18next: [Getter],
  setDefaults: [Getter],
  getDefaults: [Getter],
  setI18n: [Getter],
  getI18n: [Getter],
  composeInitialProps: [Getter],
  getInitialProps: [Getter]}
```

Not sure why `react-i18next` is behaving this way. Any thoughts @moT01, @ojeytonwilliams ?

**Comment by moT01** [comment]
I made some [mocks for react-i18next here](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/__mocks__/react-i18nextMock.js) - mostly taken from the example provided by them. It looks like we may need to mock that function - as [they say here](https://github.com/i18next/react-i18next/blob/master/example/test-jest/src/__mocks__/react-i18next.js#L45)

**Comment by ahmadabdolsaheb** [comment]
It seemed they pulled some modules and passed them on. it seems reasonable to mock the methods we need to mock and pass on the required methods from the module directly. 

I was able to shallow render the component that is being returned from layoutSelector and it has the info we need to test against.  let me know what you think.

File: client/utils/gatsby/layoutSelector.js [hunk 1]
```diff
@@ -13,21 +13,24 @@ export default function layoutSelector({ element, props }) {
...
   }
   if (/\/certification\//.test(pathname)) {
     return (
       <CertificationLayout pathname={pathname}>{element}</CertificationLayout>
     );
   }
   if (/\/guide\//.test(pathname)) {
-    console.log('Hitting guide for some reason. Need a redirect.');
+    return console.log('Hitting guide for some reason. Need a redirect.');
   }
```
Comment (raisedadead) [inline client/utils/gatsby/layoutSelector.js:29]: I think this bit can go, it's an artifact that we can remove now.

**Review by raisedadead — COMMENTED** [review]
Looks good to me, pending minor recommendations

**Review by raisedadead — APPROVED** [review]

**Pull request merged**